src/components/todo/TodoItem.tsx

```diff
@@ -3,58 +3,141 @@ import { TodoModal } from "../../models";
 import { dateConverter } from "../../utils/helper";
 import { useDispatch } from "react-redux";
 import { useState } from "react";
-import { deleteTodo, markDoneTodo } from "../../redux";
+import { deleteTodo, markDoneTodo, editTodo } from "../../store";
 import Confirm from "../shared/Confirm";
+import Input from "../shared/form/Input";
+import moment from "moment";
 interface Props {
   todoItem: TodoModal;
   id: number;
 }
 
 const TodoItem = ({ todoItem }: Props) => {
   const dispatch = useDispatch();
-  const [show, setShow] = useState(false);
-  const handleClose = () => setShow(false);
-  const handleShow = () => setShow(true);
+  const [showModal, setShowModal] = useState(false);
 
-  const handleDelete = () => {
+  // State to check edit state(if already in edit mode)
+  const [editMode, setEditMode] = useState<boolean>(false);
+
+  const [todo, setTodo] = useState({
+    id: todoItem.id,
+    todo: todoItem.todo,
+    date: todoItem.date,
+    isDone: todoItem.isDone,
+  });
+
+  const deleteTaskHandeler = () => {
     dispatch(deleteTodo(todoItem));
-    handleClose();
+    setShowModal(false);
   };
 
-  const handleDone = () => {
+  const doneTaskHandeler = () => {
     dispatch(markDoneTodo(todoItem));
   };
 
+  const editTaskHandeler = () => {
+    setEditMode(true);
+  };
+
+  const editFormDataChanger = (type: string, data: string | Date) => {
+    setTodo((todoData) => {
+      return {
+        ...todoData,
+        [type]: data,
+      };
+    });
```

Comment:
again it is having deep complication to understand, can you try what we discussed yesterday.

Reply:
Updated as discussed